### DISCUSSION TITLE
Setting up Verdaccio to listen to a specific domain

### DISCUSSION THREAD

**Discussion by NudeNULL** [verdaccio/verdaccio#3474] [Q&A] [answered]
Hello!

I am trying to use Verdaccio along with other app that is also running on the same windows machine.

But the problem is that now I have two apps that listen to the same traffic from the internet, and the only difference is the domain I am using to access them.

This is what I would like to achieve:
verdaccio.domain.com -> opens Verdaccio Web UI
otherapp.domain.com -> opens other app Web UI

How to distribute the traffic so there will be no conflicts? Any ideas would help.

Thanks!

**Comment by juanpicado** [comment, marked as answer]
You can use another port with `--listen` or use config file, check the docs.

**Reply by NudeNULL** [reply]
The apps are already listening to different ports, but correct me if I am wrong, do you suggest to use the port explicitly right after the domain, like so - verdaccio.domain.com:4873?

That's what I want to avoid, it is not clean and not comfortable for the users.

In my case, both domains will target the same port -> 443, then the router will forward traffic to one specific machine, and there, how can I distribute the traffic depending on the domain that was used?

**Reply by juanpicado** [reply]
> For example, all HTTPS traffic will be forwarded to the nginx server, and nginx will distribute the traffic onwards to the respective app depending on the domain the client use

I don't know to be honest, seems a very Nginx specific to me. I'd suggest stackoverflow probably get more expects on that area.

**Reply by NudeNULL** [reply]
Yeah, I think so, thanks for your time! If I find some answer, I will write here as well.

**Reply by juanpicado** [reply]
> The apps are already listening to different ports, but correct me if I am wrong, do you suggest to use the port explicitly right after the domain, like so - verdaccio.domain.com:4873?

No, I was suggesting to use diff ports, but seems you are doing it already.

> In my case, both domains will target the same port -> 443, then the router will forward traffic to one specific machine, and there, how can I distribute the traffic depending on the domain that was used?

I don't follow what you are trying to achieve. 

You have 

verdaccio.domain.com -> opens Verdaccio Web UI
otherapp.domain.com -> opens other app Web UI

You have two verdaccio instances?

**Reply by NudeNULL** [reply]
No, I have something else running besides Verdaccio, some other app that also have its own web interface.

I am trying to make both use secure connection with HTTPS, and both to be available from the outside, without specifying the port explicitly, which means that in result I have:

* Two different domains, that point to the same IP address through HTTPS (by default uses port 443)
* The router is forwarding all traffic that is coming through port 443 to the machine that has Verdaccio and other app running on it
* And here is the problem, I am not sure how to distribute the traffic depending on which domain was used on the client side.

Is the nginx supposed to help in this case? For example, all HTTPS traffic will be forwarded to the nginx server, and nginx will distribute the traffic onwards to the respective app depending on the domain the client used.

I see that the topic is no more related to the Verdaccio, and it is something more related to networking)) so I will close this discussion if needed.
